src/hw02/Bank.java

```diff
@@ -0,0 +1,156 @@
+package hw02;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author Single-eye
+ */
+/**
+ * Bank class replicating the function of an actual bank: its functions includes to open, modify ,
+ * and deletes accounts. *
+ */
+public class Bank {
+
+   // Holds customers accounts with the CustomerID as the key
+   private Map<Integer, Account> bankAccounts = new HashMap<>();
+
+   /**
+    * default constructor
+    */
+   public Bank() {
+
+   }
+
+   /**
+    * add a customer account after checking if an account exist or not
+    *
+    * @param in_Cust_In
+    * @param inAccount
+    */
+   public boolean addAccount(int in_Cust_In, Account inAccount) {
+
+      if (!this.queryCustAccount(in_Cust_In)) {
+
+         this.bankAccounts.put(in_Cust_In, inAccount);
+         return true;
+      } else {
+
+         return false;
+      }
+   }
+
+   /**
+    *
+    * @param in_Cust_In
+    * @return
+    */
+   public boolean deleteAccont(int in_Cust_In) {
+
+      if (this.queryCustAccount(in_Cust_In)) {
+
+         this.bankAccounts.remove(in_Cust_In);
+         return true;
+      } else {
+
+         return false;
+      }
+
+   }
+
+   /**
+    *
+    * @param in_Cust_In
+    * @param amount
+    * @return
+    * @throws AccountException
+    */
+   public Boolean Deposit(int in_Cust_In, double amount) throws AccountException {
+
+      if (this.queryCustAccount(in_Cust_In)) {
+
+         this.bankAccounts.get(in_Cust_In).withdraw(amount);
+         return true;
+      } else {
+
+         return false;
+      }
+   }
+
+   /**
+    * Adds the specified amount of money to the account balance
+    *
+    * @param amount the amount to add
+    * @return the new account balance
+    * @throws AccountException if the amount is less than $0.00
+    */
+
+   public Boolean Deposit(int in_Cust_In, double amount, boolean overdraft) throws AccountException {
+
+      if (this.queryCustAccount(in_Cust_In)) {
+
+         this.bankAccounts.get(in_Cust_In).withdraw(amount, overdraft);
+         return true;
+      } else {
+
+         return false;

```

Comment:
This might be a good place to throw an exception instead of returning false, since requesting an operation on an account that doesn't exist seems like abnormal behavior from the client. What do you think?

Reply:
updated all methods with account exception for a missing account